sampleapp/src/main/scala/de/up/hpi/informationsystems/sampleapp/dactors/Customer.scala

```diff
@@ -97,34 +97,33 @@ object Customer {
 
       case AddStoreVisit.Request(storeId: Int, time: ZonedDateTime, amount: Double, fixedDiscount: Double, varDiscount: Double) =>
         addStoreVisit(storeId, time, amount, fixedDiscount, varDiscount) match {
-          case Success(_) => sender() ! AddStoreVisit.Success(Seq.empty)
+          case Success(_) => sender() ! AddStoreVisit.Success(Relation(Seq.empty))
           case Failure(e) => sender() ! AddStoreVisit.Failure(e)
         }
 
       case Authenticate.Request(passwordHash) =>
         if (authenticate(passwordHash)) {
-          sender() ! Authenticate.Success(Seq.empty)
+          sender() ! Authenticate.Success(Relation(Seq.empty))
         } else {
           sender() ! Authenticate.Failure(AuthenticationFailedException("failed to authenticate using password"))
         }
     }
 
-    def getCustomerInfo: Try[Record] = {
+    def getCustomerInfo: Try[Relation] = {
       val rowCount = relations(CustomerInfo).records.get.size
       if (rowCount > 1) {
         throw InconsistentStateException(s"this relation was expected to contain at maximum 1 row, but contained $rowCount")
       }
-      Try(relations(CustomerInfo).records.get.head)
+      Try(Relation(relations(CustomerInfo).records.get))
```

Comment:
I see the point of re-wrapping it: To have a `TransientRelation`, which cannot be modified.

But I don't like the way it is written here. What do you think of having a method `[MutableRelation].immutable` returning a new `TransientRelation`?

Reply:
I don't feel like it's super important but it won't clutter the interface too much either so I will add it.

Reply:
Actually I'll return `Relation` instead unless you have objections to that. We moved the constructor to `Relation` anyway so and I can't come up with a reason why we should give more information on internals for this scenario.